homeassistant/components/modbus/modbus.py

```diff
@@ -236,6 +238,9 @@ def read_coils(self, unit, address, count):
             except ModbusException as exception_error:
                 self._log_error(exception_error)
                 return None
+            if type(result) in [ExceptionResponse, IllegalFunctionRequest]:
```

Comment:
Use `isinstance` to check type.
```suggestion
            if isinstance(result, (ExceptionResponse, IllegalFunctionRequest)):
```

Reply:
good catch, thanks.